src/bot/precomileInclude.jl

```diff
@@ -0,0 +1,120 @@
+
+"""
+    precompilePather(packageName::String)
+
+To get the path of precompile_packageName.jl file
+
+Written exclusively for SnoopCompile Github actions.
+# Examples
+```julia
+precompilePath, precompileFolder = precompilePather("MatLang")
+```
+"""
+function precompilePather(packageName::String)
+    return "\"../deps/SnoopCompile/precompile/precompile_$packageName.jl\"",
+    "$(pwd())/deps/SnoopCompile/precompile/"
+end
+
+precompilePather(packageName::Symbol) = precompilePather(string(packageName))
+precompilePather(packageName::Module) = precompilePather(string(packageName))
+
+################################################################
+
+function precompileRegex(precompilePath)
+    # https://stackoverflow.com/questions/3469080/match-whitespace-but-not-newlines
+    # {1,} for any number of spaces
+    c1 = Regex("#[^\\S\\r\\n]{0,}include\\($(precompilePath)\\)")
+    c2 = r"#\s{0,}_precompile_\(\)"
+    a1 = "include($precompilePath)"
+    a2 = "_precompile_()"
+    return c1, c2, a1, a2
+end
+################################################################
+
+"""
+    precompileActivator(packagePath, precompilePath)
+
+Activates precompile of a package by adding or uncommenting include() of *.jl file generated by SnoopCompile and _precompile_().
+
+Written exclusively for SnoopCompile Github actions.
+"""
+function precompileActivator(packagePath::String, precompilePath::String)
+
+    file = open(packagePath,"r")
+    packageText = Base.read(file, String)
+    close(file)
```

Comment:
You can replace this whole block with `package_text = read(filename, String)`. This applies to other places too, but I won't comment there.